### DISCUSSION TITLE
Sale and metadata variables for use in actions

### DISCUSSION THREAD

**Discussion by mixels** [gameserverapp/Platform#978] [Suggestions]
Hi!

While setting up a rudimentary shop for the first time, I couldn't find useful variables for use in actions that I've seen on other platforms. Examples come mostly from metadata about shop purchases, such as:

* Purchase date and time
* Purchase ID
* Pack / Supporter Tier purchased
* Amount and currency type paid for the order (RMT or tokens)

Variables like these can help a team manage purchases better in ways like better notifying a player which purchase is being filled and helping the team log more relevant and useful information to Discord for simplified resolution in case an automatic delivery fails for some reason.

**Comment by goosenation** [comment]
Would be nice to see more variables interpolated in text and embed messages on Discord. See image.

Variables that come to mind are:

1. {amount}
2. ... *will update in future*

![image](https://github.com/user-attachments/assets/015454e9-2539-46ee-a577-f4b260912182)

**Reply by mvaessen** [reply]
This will be added separately, but part of the same release

**Comment by mvaessen** [comment]
Thanks for the suggestion!

Adding the following options:

For Shop packs:
- `{variable.purchase_date}` - *3 Mar @ 14:28*
- `{variable.purchase_amount}` - *5*
- `{variable.purchase_currency} `- *tokens*
- `{variable.transaction_description}` - *Order: Awesome shop item*
- `{variable.item_name}` - *Awesome shop item*

For Supporter Tiers (incl. Patreon):
- `{variable.purchase_date}` - *3 Mar @ 14:28*
- `{variable.purchase_amount}` - *5.00*
- `{variable.purchase_currency} `- *€*
- `{variable.purchase_type} `- *One-time*
- `{variable.gateway}` - *PayPal*
- `{variable.gateway_refid} `- *0WU229153V441691T*
- `{variable.item_name}` - *Bestest single purchase*

If you're looking for other variables directly related to the purchase (not the actions inside the purchased item) please let me know
